build-tools/packages/build-tools/src/genMonoRepoPackageJson/lib.ts

```diff
@@ -4,121 +4,82 @@
  */
 
 import path from "path";
-import {
-    MonoRepo,
-    readJsonAsync,
-    writeFileAsync,
-    Package,
-    Logger,
-} from "@fluidframework/build-tools";
+import { commonOptions } from "../common/commonOptions";
+import { Logger } from "../common/logging";
+import { MonoRepo } from "../common/monoRepo";
+import { Package } from "../common/npmPackage";
+import { readJsonAsync, writeFileAsync } from "../common/utils";
 
 function format(n: number) {
     return n.toString().padStart(4);
 }
 
-/**
- *
- * @param monoRepo - MonoRepo
- * @param repoPackageJson - any
- * @param logger - Logger
- * @returns - lerna doesn't distingish between dependencies vs devDependencies, this function
- * will use the lerna-package-lock.json and patch up the "dev" field in the dependencies and
- * output it to repo-package-lock.json
- */
-
-async function generateMonoRepoPackageLockJson(
-    monoRepo: MonoRepo,
-    repoPackageJson: any,
-    logger: Logger,
-) {
+async function generateMonoRepoPackageLockJson(monoRepo: MonoRepo, repoPackageJson: any, log?: Logger) {
     // Patching the package-lock file
-    const repoPackageLockJson = await readJsonAsync(
-        path.join(monoRepo.repoPath, "lerna-package-lock.json"),
-    );
+    const repoPackageLockJson = await readJsonAsync(path.join(monoRepo.repoPath, "lerna-package-lock.json"));
 
     let totalDevCount = 0;
     let topLevelDevCount = 0;
 
     const setDev = (item: any) => {
         totalDevCount++;
         item.dev = true;
-        if (item.dependencies !== undefined) {
-            return;
-        }
-
-        // eslint-disable-next-line guard-for-in
+        if (!item.dependencies) { return; }
         for (const dep in item.dependencies) {
             setDev(item.dependencies[dep]);
         }
-    };
+    }
 
     // Assume all of them are dev dependencies
-    // eslint-disable-next-line guard-for-in
     for (const dep in repoPackageLockJson.dependencies) {
         topLevelDevCount++;
         setDev(repoPackageLockJson.dependencies[dep]);
     }
-
     const totalCount = totalDevCount;
     const topLevelTotalCount = topLevelDevCount;
 
     const markNonDev = (name: string, topRef: string, item: any, refStack: any[]) => {
         totalDevCount--;
         delete item.dev;
         refStack.push(item);
-
-        if (item.dependencies !== undefined) {
+        if (item.dependencies) {
```

Comment:
Nit: I think our stricter linter configs prefer explicit `!== undefined`, don't they? Should we use that here?

Reply:
Actually, this was a bug. The copied code used `!== undefined`, but that was too narrow. I added a comment explaining this.